specification/monitor/resource-manager/Microsoft.Insights/stable/2022-06-15/scheduledQueryRule_API.json

```diff
@@ -0,0 +1,671 @@
+        },
+        "overrideQueryTimeRange": {
+          "type": "string",
+          "format": "duration",
+          "description": "If specified then overrides the query time range (default is WindowSize*NumberOfEvaluationPeriods). Relevant only for rules of the kind LogAlert."
+        },
+        "targetResourceTypes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "List of resource type of the target resource(s) on which the alert is created/updated. For example if the scope is a resource group and targetResourceTypes is Microsoft.Compute/virtualMachines, then a different alert will be fired for each virtual machine in the resource group which meet the alert criteria. Relevant only for rules of the kind LogAlert"
+        },
+        "criteria": {
+          "$ref": "#/definitions/ScheduledQueryRuleCriteria",
+          "description": "The rule criteria that defines the conditions of the scheduled query rule."
+        },
+        "muteActionsDuration": {
+          "type": "string",
+          "format": "duration",
+          "description": "Mute actions for the chosen period of time (in ISO 8601 duration format) after the alert is fired. Relevant only for rules of the kind LogAlert."
+        },
+        "actions": {
+          "type": "object",
+          "$ref": "#/definitions/Actions",
+          "description": "Actions to invoke when the alert fires."
+        },
+        "isWorkspaceAlertsStorageConfigured": {
+          "type": "boolean",
+          "readOnly": true,
+          "description": "The flag which indicates whether this scheduled query rule has been configured to be stored in the customer's storage. The default is false."
+        },
+        "checkWorkspaceAlertsStorageConfigured": {
+          "type": "boolean",
+          "description": "The flag which indicates whether this scheduled query rule should be stored in the customer's storage. The default is false. Relevant only for rules of the kind LogAlert."
+        },
+        "skipQueryValidation": {
+          "type": "boolean",
+          "description": "The flag which indicates whether the provided query should be validated or not. The default is false. Relevant only for rules of the kind LogAlert."
+        },
+        "autoMitigate": {
+          "type": "boolean",
+          "description": "The flag that indicates whether the alert should be automatically resolved or not. The default is true. Relevant only for rules of the kind LogAlert."
+        }
+      },
+      "description": "scheduled query rule Definition"
+    },
+    "ScheduledQueryRuleResource": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/TrackedResource"
```

Comment:
Hi @avjerafi , 

I would suggest fix the incompatible issues in this PR.
I can find that problem comes from Line 441  and Line 656 where refer to common-types.  Would you try try to not  ref  to common-types here?
I see other specs in this tag defines some Resource, maybe you can reuse these definitions?

> Thank you @changlong-liu.
> 
> I've applied the flag as you directed and the "SDK Track2 Validation" error was suppressed.
> 
> However, LintDiff validation error is still not suppressed, even though they both refer to the same issue, as far as I can tell. I cannot completely resolve this within the scope of this PR because it involves cooperation with many teams in order to align all the types used in all the different resource types.
> 
> This effort will certainly be part of the next SDK release, which we are working on these days in parallel to publishing this new api version.
> 
> Please advise. Thanks!

Reply:
Hi @changlong-liu,

- For "Resource": I've removed the "allOf" reference and inlined all the fields inside the local definition "ScheduledQueryRuleResource".
- For "ErrorResponse": I've duplicated the definition from common-types as a new local definition, renamed as the unique name "ErrorResponseDetails" in order to avoid any conflicts.

Despite that, I still get this LintDiff validation error: "model name with non-identical definitions" but I can't find which definition this error relates to.

All the local validations done on my machine passed.

Please advise.
Thanks!
Please advise.
Thanks you!

Reply:
hI @avjerafi ,
Detail errors can be found at https://dev.azure.com/azure-sdk/590cfd2a-581c-4dcb-a12e-6568ce786175/_apis/build/builds/1708473/logs/323.
Now it's for duplicated definition of 'ErrorAdditionalInfo'.

And for the LintDiff error, you should can get reproduced in local with below command:
~~~
autorest --v2 --validation --azure-validator --semantic-validator=false --model-validator=false --message-format=json --openapi-type=arm --use=@microsoft.azure/classic-openapi-validator@1.2.4 --use=@microsoft.azure/openapi-validator@1.13.0 --tag=package-2022-06 [PATH-TO-SWAGGER-REPO]/specification/monitor/resource-manager/readme.md
~~~

Reply:
Hi @changlong-liu,
Thank you for your guidance!
I've fixed the "ErrorAdditionalInfo" reference by renaming it as well.
LintDiff validation now finally passes.
Is the PR now approvable?
Thanks!